plugins/cairo-lang-macro/src/lib.rs

```diff
@@ -3,13 +3,93 @@ pub use cairo_lang_macro_attributes::*;
 pub use linkme::distributed_slice;
 
 use cairo_lang_macro_stable::ffi::StableSlice;
-use cairo_lang_macro_stable::{StableAuxData, StableProcMacroResult};
+use cairo_lang_macro_stable::{StableAuxData, StableExpansionsList, StableProcMacroResult};
+use once_cell::sync::Lazy;
+use std::collections::HashMap;
+use std::ffi::{c_char, CStr};
 use std::slice;
 
 mod types;
 
 pub use types::*;
 
+#[derive(Clone)]
+pub struct ExpansionDefinition {
+    pub name: &'static str,
+    pub kind: ExpansionKind,
+    pub fun: ExpansionFunc,
+}
+
+type ExpansionFunc = fn(TokenStream) -> ProcMacroResult;
+
+/// Distributed slice for storing procedural macro code expansion capabilities.
+///
+/// Each element denotes name of the macro, and the expand function pointer.
+#[distributed_slice]
+pub static MACRO_DEFINITIONS_SLICE: [ExpansionDefinition];
+
+static EXPAND_CALLBACK: Lazy<HashMap<String, ExpansionFunc>> = Lazy::new(|| {
+    MACRO_DEFINITIONS_SLICE
+        .iter()
+        .map(|m| (m.name.to_string(), m.fun))
+        .collect()
+});
```

Comment:
Do you really need to invest code into creating this hash map? Why not just find the item linearly in the source vector?

Anyway, I bet that for the amounts of items this collection will get, searching `vec` will be faster than making a hashmap and looking up there